src/dosext/mfs/mfs.c

```diff
@@ -4294,6 +4299,47 @@ static int dos_fs_redirect(struct vm86_regs *state, char *stk)
       break;
     }
 
+    case GET_LARGE_DISK_SPACE: /* 0xa3 */
+    {
+      cds_t tcds = Addr(state, es, edi);
+      char *name = cds_current_path(tcds);
+      uint32_t avail, total;
+      uint16_t blocksize;
+      int dd;
+      struct statfs fsbuf;
+
+      Debug0((dbg_fd, "Get Large Disk Space(INT2F/11a3)\n"));
+
+      if (!get_drive_from_path(name, &dd)) {
+        Debug0((dbg_fd, "Bad drive name '%s'\n", name));
+        break;
+      }
+
+      if (!drives[dd].root) {
+        Debug0((dbg_fd, "Drive not ours\n"));
+        break;
+      }
+
+      if (statfs(drives[dd].root, &fsbuf) == -1) {
+        Debug0((dbg_fd, "Can't stat root path '%s'\n", strerror(errno)));
+        break;
```

Comment:
Since we have by this point determined that it's our drive, should this return an error rather than chain to the next redirector?

Reply:
Its your code, why you ask me? :)
If you want to return an error, please
open a PR. Obviously an error from
statfs() is unexpected, so the range
of options is high. From ignoring the
ret value completely to an assert.

Reply:
> Its your code, why you ask me? :)

I value your opinion.